mateog-assign1/crypto.py

```diff
@@ -0,0 +1,322 @@
+        for i in range(0, len(plaintext)):
+            frag_num = int(i / frag_len)
+            if frag_num % 2 == 0:
+                rails[i % frag_len] += plaintext[i]
+            else:
+                rails[-(1 + i % frag_len)] += plaintext[i]
+
+        ciphertext = ""
+        for r in rails:
+            ciphertext += r
+
+        return ciphertext
+
+
+def decrypt_railfence(ciphertext, num_rails):
+        """
+        Encrypts plaintext using a railfence cipher.
+        Add more implementation details here.
+        """
+        ciphertext = ciphertext.upper()
+        frag_len = num_rails - 1
+        rails = [""] * num_rails
+        num_full_frags = int(len(ciphertext) / frag_len)
+        len_last_frag = len(ciphertext) % frag_len
+        for i in range(0, num_rails):
+
+            # TOP RAIL
+            if i == 0:
+                num_peaks = int(num_full_frags / 2)
+                if num_full_frags % 2 != 0 or len_last_frag > 0: # If extra letter from full/part fragment
+                    num_peaks += 1
+                rails[i] = ['-'] * num_peaks
+
+            # BOTTOM RAIL
+            elif i == num_rails - 1:
+                num_troughs = int(num_full_frags / 2)
+                if num_full_frags % 2 != 0 and len_last_frag > 0: # If extra letter from full/part fragment
+                    num_troughs += 1
+                rails[i] = ['-'] * num_troughs
+
+            # MIDDLE RAILS
+            else:
+                num_chars = num_full_frags
+                if num_full_frags % 2 == 0 and len_last_frag > i: # same  as ^^
+                    num_chars += 1
+                elif num_full_frags % 2 != 0 and num_rails - len_last_frag <= i: # same  as ^^
+                    num_chars += 1
+                rails[i] = ['-'] * num_chars
+
+        # CONSTRUCT RAILS
+        spacing = frag_len * 2
+        for i in range(0, num_rails):
+            if i == 0 or i == num_rails - 1:
+                rails[i] = ciphertext[::spacing]
+            else:
+                rails[i][::2] = ciphertext[::spacing]
+                odd_index_char_start = i + ((frag_len - i) * 2)
+                rails[i][1::2] = ciphertext[odd_index_char_start::spacing]
+
+        # CONSTRUCT FRAGMENTS
+        num_frags = num_full_frags
+        if num_full_frags > 0:
+            num_frags += 1
+        frags = [""] * num_frags
+        for i in range(0, num_frags):
+            startIndex = i * frag_len
+            frags[i] = ciphertext[startIndex:startIndex + frag_len]
+
+        # CONVERT TO PLAINTEXT
+        plaintext = list(ciphertext)
+        for i in range(0, len(ciphertext)):
+            frag_num = int(i / frag_len)
+            if frag_num % 2 == 0:
+                plaintext[i] = rails[i % frag_len][0]
+                rails[i % frag_len] = rails[i % frag_len][1:]
+            else:
+                plaintext[i] = rails[-(1 + i % frag_len)][0]
+                rails[-(1 + i % frag_len)] = rails[-(1 + i % frag_len)][1:]
```

Comment:
It's not super clear here why you're doing `-(1+i%frag_len)`...would appreciate a comment.
In general, I think I get the gist of it though...